src/univariate/discrete/binomial.jl

```diff
@@ -105,6 +105,24 @@ function entropy(d::Binomial; approx::Bool=false)
     end
 end
 
+function kldivergence(p::Binomial, q::Binomial; kwargs...)
+    np = ntrials(p)
+    nq = ntrials(q)
+    succp = succprob(p)
+    succq = succprob(q)
+    res = np * kldivergence(Bernoulli{typeof(succp)}(succp), Bernoulli{typeof(succq)}(succq))
+    if np == nq
+        iszero(np) && return zero(res)
+        return res
+    elseif np > nq
+        return oftype(res, Inf)
+    else
+        # pull some terms out of the expectation to make this more efficient:
+        res += logfactorial(np) - logfactorial(nq) - (nq - np) * log1p(-succq)
+        res += expectation(k -> logfactorial(nq - k) - logfactorial(np - k), p)
```

Comment:
I thought pulling out the constant terms is an obvious optimization, and it is faster than the generic fallback.
I presume the explicit sum would be even faster, since all binomial coefficients would only need to be calculated once.

Reply:
I think let's keep it this way for now and perform optimizations/changes in follow-up PRs.